### ISSUE TITLE
Const reassignment in production build

### ISSUE THREAD

**Issue by richinfante** [webpack/webpack#13022]
# Bug report

**What is the current behavior?**

Webpack generates code using a `const` variable and then attempts to re-assign it. I suppose this could be due to the vue dependencies I'm using - but this code very much appears to be generated by webpack itself (or at least, [half of it is](https://github.com/webpack/webpack/blob/fd056e27f477e364c16d1b09b39905fce28c13ca/lib/dependencies/HarmonyExportExpressionDependency.js#L140), and then the `libraryExport` option attempts to overwrite it).

It seems like this shouldn't be creating a block-scoped variable here at all, and assigning it to the global variable for the named library export.

**If the current behavior is a bug, please provide the steps to reproduce.**

Repo with minimal reproduction:
https://github.com/richinfante/webpack-bug-repro

I've disabled the terser plugin in the test repo so output isn't minified & optimized.

In short, running a production build generates the following code, which yields a syntax error in browsers:
```js
/* harmony default export */ const empty_component = (component.exports);
empty_component = __webpack_exports__.default;
```

**What is the expected behavior?**

Webpack shouldn't generate code that tries to re-assign a const variable.

**Other relevant information:**
webpack version:  5.28.0
Node.js version: v14.16.0
Operating System: Arch Linux (5.11.8-arch1-1)
Additional tools: Vue 2.6.12

**Comment by richinfante** [comment]
I've done a little more digging on this - setting `output.environment.const` to false fixes the const syntax error, but the build is still broken and the export is set to `undefined` for the affected modules. This might be more related to the handling of library exports and the `const` issue is just a side-effect.

I added an example of a [bad build](https://github.com/richinfante/webpack-bug-repro/blob/main/diffs/bad_compile.js) and a known [good build](https://github.com/richinfante/webpack-bug-repro/blob/main/diffs/good_compile.js) of the same module into the `diff` folder in the repo - They're minified b/c I pulled them off one of our build servers, but it's the best I have at the moment to easily compare

**Comment by alexander-akait** [comment]
Yep, bug on webpack side

**Comment by alexander-akait** [comment]
Marked as `critical-bug`, we will try to fix it in near future

**Comment by sokra** [comment]
as workaround: rename your `empty_component.vue` file to something else.

**Comment by hmjhgygfgrfd** [comment]
D

**Comment by alexander-akait** [comment]
bump
